src/Scripting/Core/ScriptOptions.cs

```diff
@@ -282,5 +301,17 @@ private static MetadataReference CreateReferenceFromAssembly(Assembly assembly)
         /// <exception cref="ArgumentNullException"><paramref name="imports"/> is null or contains a null reference.</exception>
         public ScriptOptions AddImports(params string[] imports) =>
             AddImports((IEnumerable<string>)imports);
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with debugging information enabled.
+        /// </summary>
+        public ScriptOptions WithEmitDebugInformation(bool emitDebugInformation) =>
+            emitDebugInformation == EmitDebugInformation ? this : new ScriptOptions(this) { EmitDebugInformation = emitDebugInformation };
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="FileEncoding"/>.
+        /// </summary>
+        public ScriptOptions WithFileEncoding(Encoding encoding) => 
```

Comment:
No, we keep the ```FileEncoding``` property.  If the encoding is not passed to SourceText the detection is attempted (using BOM) and if we can't detect we default to UTF8 with no BOM. If there is no BOM the encoding needs to be specified, which is what FileEncoding would be used for. You can look at the implementation of http://source.roslyn.io/#Microsoft.CodeAnalysis/Text/SourceText.cs,3687bd0c8922a81f to see what it's doing. Note the exceptions it may throw. We should pass them on the to the caller of the script APIs that take Stream and document them.